app/controllers/amr_heatmap_controller.rb

```diff
@@ -58,4 +63,174 @@ def amr_counts
+    }
+
+    # Some drug resistances are only listed in the gene family entry
+    if remote_properties["drugClass"].empty?
+      drug_classes = []
+      remote_properties["geneFamily"].each do |node|
+        gene_family_accession = node.get_attribute("rdf:resource").split('_')[1]
+        drug_classes.concat(get_drug_classes(gene_family_accession, xml_dom))
+      end
+      remote_properties["drugClass"] = drug_classes
+    end
+
+    local_properties = {
+      "synonyms" => matching_entry.xpath("./oboInOwl:hasExactSynonym"),
+      "publications" => matching_entry.at_xpath("./following-sibling::owl:Axiom").xpath("./oboInOwl:hasDbXref")
+    }
+
+    local_properties.each do |ontology_key, property_nodes|
+      collection = []
+      property_nodes.each do |node|
+        collection.push(node.content)
+      end
+      ontology_information[ontology_key] = collection
+    end
+
+    remote_properties.each do |ontology_key, property_nodes|
+      collection = []
+      property_nodes.each do |node|
+        accession = node.get_attribute("rdf:resource").split('_')[1]
+        accession_class = xml_dom.at_xpath(".//owl:Class[@rdf:about='http://purl.obolibrary.org/obo/ARO_#{accession}']")
+        label = accession_class.at_xpath("./rdfs:label").content
+        description = accession_class.at_xpath("./obo:IAO_0000115").content
+        entry = { "label" => label, "description" => description }
+        collection.push(entry)
+      end
+      ontology_information[ontology_key] = collection
+    end
+    return ontology_information
+  end
+
+  def get_drug_classes(accession, xml_dom)
+    accession_class = xml_dom.at_xpath(".//owl:Class[@rdf:about='http://purl.obolibrary.org/obo/ARO_#{accession}']")
+    drug_class_node_markers = accession_class.xpath("./rdfs:subClassOf/owl:Restriction/owl:onProperty[@rdf:resource='http://purl.obolibrary.org/obo/RO#_confers_resistance_to_drug']")
+    drug_family_node_markers = accession_class.xpath("./rdfs:subClassOf/owl:Restriction/owl:onProperty[@rdf:resource='http://purl.obolibrary.org/obo/RO#_confers_resistance_to']")
+    drug_class_nodes = []
+    drug_class_node_markers.each do |node_marker|
+      drug_class_nodes.push(node_marker.at_xpath("./following-sibling::owl:someValuesFrom"))
+    end
+    drug_family_node_markers.each do |node_marker|
+      drug_class_nodes.push(node_marker.at_xpath("./following-sibling::owl:someValuesFrom"))
+    end
+    return drug_class_nodes
+  end
+
+  def get_publication_name(pmid)
+    pubmed_uri = URI(URL_PUBMED + pmid)
+    response = Net::HTTP.get_response(pubmed_uri)
+    unless response.is_a?(Net::HTTPSuccess)
+      return pmid
+    end
+    publication_page = Nokogiri::HTML(response.body)
+    abstract_node = publication_page.at_xpath(".//div[@class='rprt abstract']")
+    authors = abstract_node.at_xpath("./div[@class='auths']").content
+    title = abstract_node.at_xpath("./h1").content
+    pub_name = authors + " " + title + " (PMID #{pmid})"
```

Comment:
you should use string interpolation for the whole line like `#{authors} #{title}`..